syntax = "proto3";

package word_counter;

service Counter {
    rpc Count (WordCountRequest) returns (WordCountResponse);
}

message WordCountRequest {
    string word = 1;        // 关键词
    string file_name = 2;   // 文本文件名
}

message WordCountResponse {
    int64 count = 1;        // 关键词出现次数
    int64 status_code = 254; // 状态码
    string status_message = 255; // 状态消息
    string log_id = 256;    // 日志ID
}
